### PR TITLE
[mypy] add stubs for color functions

### PR DESCRIPTION
**Pull request by picnixz**
This adds stubs for color functions declared in ``sphinx.util.console``.

This also removes ``# type: ignore[attr-defined]`` introduced due to the lack of stubs and apply `ruff format` to `sphinx.util.console` (since there are a lot of files in `sphinx/util` that are ignored by `ruff format` I did not explicitly removed it from the exclusion list).

### CODE REVIEW

File: sphinx/util/console.py [hunk 1]
```diff
@@ -20,13 +47,15 @@
 
 _CSI = re.escape('\x1b[')  # 'ESC [': Control Sequence Introducer
 _ansi_re: re.Pattern[str] = re.compile(
-    _CSI + r"""
+    _CSI
+    + r"""
```
Comment (chrisjsewell) [inline sphinx/util/console.py:51]: Maybe this file could now be removed from the `ruff format` exclude list, as it looks like you formatted it directly?

**Review by chrisjsewell — APPROVED** [review]
All good thanks, just the one comment

File: sphinx/util/console.py (hunk 1 above)
Comment (picnixz) [reply sphinx/util/console.py:51]: Well, the issue is that the exclusion list excludes ``sphinx/util/**/*`` so.. is there a way to exclude this file from the exclusion list but keep every other files excluded?

Comment (chrisjsewell) [reply sphinx/util/console.py:51]: Well I created the current list with a little `Path.glob` use, I'm sure you can work out how to do the same 😉 , i.e to "expand" `sphinx/util/**/*`

Comment (picnixz) [reply sphinx/util/console.py:51]: Ah, yes. I'll make another PR to expand everything (it's easier like that in the future). I was wondering whether there was an option in ruff for that.

**Pull request merged**